internal/rukpak/util/hash.go

```diff
@@ -0,0 +1,39 @@
+package util
+
+import (
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"math/big"
+)
+
+// DeepHashObject writes specified object to hash using the spew library
+// which follows pointers and prints actual values of the nested objects
+// ensuring the hash does not change when a pointer changes.
+func DeepHashObject(obj interface{}) (string, error) {
```

Comment:
TODO (follow up): We can have this for now, but just wondering if we need to ever use it anywhere to hash the entire spec of the object. The linter would complain in the future anyway, we can take care as the design proceeds.